groovy-mode.el

```diff
@@ -227,6 +228,9 @@ The function name is the second group in the regexp.")
          )
        'symbols)
      . font-lock-keyword-face)
+    ;; Only highlight as keyword if not the annotation.
+    (,(rx (or bol (not (any "@"))) (group "interface"))
+     . font-lock-keyword-face)
```

Comment:
Close, but not quite. This means that we highlight `xinterface` here:

```groovy
def xinterfacex = 1
```

Emacs applies `font-lock-defaults` in order, so you just need `@interface` first:

```emacs-lisp
(defvar groovy-font-lock-keywords
  `((,(regexp-opt
       ;; Annotations are defined with the @interface, which is a keyword:
       ;; http://groovy-lang.org/objectorientation.html#_annotation
       ;; but it's nicer to use annotation highlighting.
       '("@interface")
       'symbols)
     . font-lock-keyword-face)
    (,(regexp-opt
       ;; http://docs.groovy-lang.org/latest/html/documentation/#_keywords
       '("as"
         "assert"
         "break"
         "case"
         "catch"
         "class"
         "const"
         "continue"
         "def"
         "default"
         "do"
         "else"
         "enum"
         "extends"
         "finally"
         "for"
         "goto"
         "if"
         "implements"
         "import"
         "in"
         "instanceof"
         ;; handle this below
         ;;"interface"
         "new"
         "package"
         "return"
         "super"
         "switch"
         "this"
         "throw"
         "throws"
         "trait"
         "try"
         "while"
         ;; Other strings that we want to highlight as keywords.
         "abstract"
         "final"
         "native"
         "private"
         "protected"
         "public"
         "static"
         "synchronized"
         )
       'symbols)
     . font-lock-keyword-face)
    ;; Annotations
    (,(rx "@" symbol-start (+ (or (syntax word) (syntax symbol))) symbol-end)
     . groovy-annotation-face)
    ;; ... and the rest of the patterns as before.
    ))
```